Helper/Cart.php

```diff
@@ -1754,7 +1754,7 @@ public function buildCartFromQuote($quote, $immutableQuote, $items, $placeOrderP
         if ($totalAmount < 0) {
             $totalAmount = 0;
             foreach ($cart['discounts'] as &$discount) {
-                $discount['type'] = 'fixed_amount';
```

Comment:
This change does not seem (backward) compatible with the current flow. You should probably leave the `type` field as well. Or verify the impact and be 100% sure that `discount_type` and `type` are treated as synonyms on Bolt side. There's functionality around this that needs to stay - when the amount of discount / gift certificate / store credit / reward points is above the order total.

Reply:
Good catch here @danac-gs and I agree. We can't guarantee all merchants who auto-update to this version are going to work with their CSM to properly configure the new endpoint.

Reply:
@danac-gs @matt-thomason I have discussed with @melvin0008 , and for now we do need both of the two properties, so `type` is for v1/merchant/order, `discount_type ` is for discount.apply and v2/update-cart

Then I have fixed this, please review again. Thanks